### PR TITLE
error handling

### PR DESCRIPTION
**Pull request by mateusztheglue**

### CODE REVIEW

**Comment by theglue-cicd** [comment]
:star: PR built and available in a preview environment **the-glue-nnvp-backend-pr-9** [here](http://nnvp-backend.jx-the-glue-nnvp-backend-pr-9.cluster.development.nnvp.thegluesandbox.com)

File: src/main/java/com/theglue/nnvp/controller/exceptionhandler/RestExceptionHandler.java [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+    if (exception.getErrorMessage() != null) {
+      return createErrorResponse(
+          REQUEST_VALIDATION_FAILED_MSG,
+          HttpStatus.BAD_REQUEST,
+          ErrorItem.builder()
+              .code(exception.getErrorMessage().getCode())
+              .message(exception.getErrorMessage().getMessage())
+              .build());
+    } else {
+      return createErrorResponse(
```
Comment (eryk7b) [inline src/main/java/com/theglue/nnvp/controller/exceptionhandler/RestExceptionHandler.java]: we don't need else if there's if() {return} else {return}

File: src/main/java/com/theglue/nnvp/controller/exceptionhandler/RestExceptionHandler.java [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+          INVALID_REQUEST_MSG,
+          HttpStatus.BAD_REQUEST,
+          ErrorItem.builder()
+              .message(
+                  INVALID_PROPERTY_MSG
+                      + ": "
+                      + ((PropertyBindingException) cause).getPropertyName())
+              .build());
+    } else if (cause instanceof InvalidFormatException) {
+      return invalidFormatExceptionHandler((InvalidFormatException) cause);
```
Comment (eryk7b) [inline src/main/java/com/theglue/nnvp/controller/exceptionhandler/RestExceptionHandler.java]: same here (if-return), just could improve readability

File: src/main/java/com/theglue/nnvp/controller/exceptionhandler/ErrorMessage.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package com.theglue.nnvp.controller.exceptionhandler;
+
+import com.theglue.common.model.enums.SearchableEnum;
+
+import lombok.Getter;
+
+@Getter
+public enum ErrorMessage implements SearchableEnum {
+  ;
```
Comment (eryk7b) [inline src/main/java/com/theglue/nnvp/controller/exceptionhandler/ErrorMessage.java]: ; to delete

**Comment by theglue-cicd** [comment]
:star: PR built and available in a preview environment **the-glue-nnvp-backend-pr-9** [here](http://nnvp-backend.jx-the-glue-nnvp-backend-pr-9.cluster.development.nnvp.thegluesandbox.com)

**Pull request merged**